reports-scheduler/src/test/kotlin/com/amazon/opendistroforelasticsearch/reportsscheduler/rest/ReportInstanceIT.kt

```diff
@@ -0,0 +1,290 @@
+            "$BASE_REPORTS_URI/instance/$dummyInstanceId",
+            updateStatus,
+            RestStatus.NOT_FOUND.status
+        )
+        validateErrorResponse(reportDefinitionUpdateResponse, RestStatus.NOT_FOUND.status)
+    }
+
+    fun `test update report instance status with invalid update status`() {
+        val reportDefinitionRequest = constructReportDefinitionRequest()
+        val reportDefinitionResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/definition",
+            reportDefinitionRequest,
+            RestStatus.OK.status
+        )
+
+        val reportDefinitionId = reportDefinitionResponse.get("reportDefinitionId").asString
+        Assert.assertNotNull("reportDefinitionId should be generated", reportDefinitionId)
+        Thread.sleep(100)
+        val onDemandRequest = """
+            {}
+        """.trimIndent()
+        val onDemandResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/on_demand/$reportDefinitionId",
+            onDemandRequest,
+            200
+        )
+
+        val updateStatus = """
+            {
+                "invalidStatus":"Success",
+                "statusText":"Operation completed"
+            }
+        """.trimIndent()
+        val reportInstance = onDemandResponse.get("reportInstance").asJsonObject
+        val reportInstanceId = reportInstance.get("id").asString
+
+        val reportDefinitionUpdateResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/instance/$reportInstanceId",
+            updateStatus,
+            RestStatus.BAD_REQUEST.status
+        )
+        val error = reportDefinitionUpdateResponse.get("error").asJsonObject
+        val type = error.get("type").asString
+        Assert.assertEquals(
+            "illegal_argument_exception",
+            type
+        )
+    }
+
+    fun `test listing all report instances`() {
+        val reportDefinitionRequest = constructReportDefinitionRequest()
+        val reportDefinitionResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/definition",
+            reportDefinitionRequest,
+            RestStatus.OK.status
+        )
+
+        val reportDefinitionId = reportDefinitionResponse.get("reportDefinitionId").asString
+        Assert.assertNotNull("reportDefinitionId should be generated", reportDefinitionId)
+        Thread.sleep(100)
+        val onDemandRequest = """
+            {}
+        """.trimIndent()
+        val onDemandResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/on_demand/$reportDefinitionId",
+            onDemandRequest,
+            200
+        )
+        val reportInstance = onDemandResponse.get("reportInstance").asJsonObject
+        val reportInstanceId = reportInstance.get("id").asString
+
+        val addReportDefinition = """
```

Comment:
can it be simplified as `val addReportDefinition = constructReportDefinitionRequest(name = "new report definition")`?

Reply:
Yes, good catch